laevatein/src/main/java/com/laevatein/internal/utils/MediaStoreUtils.java

```diff
@@ -0,0 +1,343 @@
+            values.put(MediaStore.Images.Media.DESCRIPTION, "mixi Photo");
+            values.put(MediaStore.Images.Media.ORIENTATION,
+                    ExifInterfaceUtils.getExifOrientation(file.getAbsolutePath()));
+            long date = ExifInterfaceUtils.getExifDateTimeInMillis(file.getAbsolutePath());
+            if (date != -1)
+                values.put(MediaStore.Images.Media.DATE_TAKEN, date);
+
+            Uri imageUri = mContext.getContentResolver().insert(
+                    MediaStore.Images.Media.EXTERNAL_CONTENT_URI, values);
+
+            FileOutputStream fos = (FileOutputStream) mContext.getContentResolver().openOutputStream(
+                    imageUri);
+            FileInputStream fis = new FileInputStream(file);
+            copyFileStream(fis, fos);
+            fos.close();
+            fis.close();
+
+            generateThumbnails(ContentUris.parseId(imageUri));
+
+            return imageUri;
+        } catch (Exception e) {
+            Log.w(TAG, "cannot insert", e);
+            return null;
+        }
+    }
+
+    /**
+     * Request to update latest photos onto media store.
+     */
+    private void updateLatestPhotos() {
+        // retrieve the newest five images
+        Cursor c = MediaStore.Images.Media.query(mContext.getContentResolver(),
+                MediaStore.Images.Media.EXTERNAL_CONTENT_URI, new String[]{
+                        MediaStore.Images.ImageColumns._ID,
+                        MediaStore.Images.ImageColumns.DATE_TAKEN,
+                        MediaStore.Images.ImageColumns.SIZE,
+                }, null, null, MediaStore.Images.ImageColumns.DATE_ADDED + " DESC");
+        if (c != null) {
+            try {
+                int count = 0;
+                mRecentlyUpdatedPhotos = new ArrayList<PhotoContent>();
+                while (c.moveToNext()) {
+                    PhotoContent item = new PhotoContent();
+                    item.id = c.getLong(0);
+                    item.taken = c.getLong(1);
+                    item.size = c.getInt(2);
+                    mRecentlyUpdatedPhotos.add(item);
+                    if (++count > 5)
+                        break;
+                }
+            } finally {
+                c.close();
+            }
+        }
+    }
+
+    /**
+     * Create thumbnail images for a specified photo image.
+     *
+     * @param imageId referes to a photo image.
+     */
+    private void generateThumbnails(long imageId) {
+        try {
+            MediaStore.Images.Thumbnails.getThumbnail(mContext.getContentResolver(), imageId,
+                    MediaStore.Images.Thumbnails.MINI_KIND, null);
+        } catch (NullPointerException e) {
+            // some MediaStores throws NPE, just ignore the result
+        }
+    }
+
+    /**
+     * Make output file uri based on a external storage directory.
+     *
+     * @return a file
+     */
+    @TargetApi(Build.VERSION_CODES.FROYO)
+    private File getOutputFile() {
```

Comment:
Renamed `getOutputFileUri()` to `getOutputFile()`.
see: https://github.com/mixi-inc/Android-Device-Compatibility/blob/master/AndroidDeviceCompatibility/src/main/java/jp/mixi/compatibility/android/provider/MediaStoreCompat.java#L325